stellarobservatory/quorum_lachowski_test.py

```diff
@@ -71,3 +76,34 @@ def test_has_quorum_intersection_true_sccs():
         5: [{2, 5}]
     }
     assert has_quorum_intersection(slices_by_node) is True
+
+
+def test_is_quorum():
+    """Test is_quorum()"""
+    slices_by_node = {
+        1: [{1, 2}, {1, 3}, {1, 2, 3}],
+        2: [{1, 2}, {1, 4}],
+        3: [{1, 2, 3, 4}]
+    }
+    assert is_quorum(slices_by_node, {1, 2}) is True
+    assert is_quorum(slices_by_node, {1, 2, 3}) is False
+
+
+def test_enumerate_quorums():
+    """Test enumerate_quorums()"""
+    slices_by_node = {
+        1: [{1, 2, 3, 7}],
+        2: [{1, 2, 3, 7}],
+        3: [{1, 2, 3, 7}],
+        4: [{4, 5, 6, 7}],
+        5: [{4, 5, 6, 7}],
+        6: [{4, 5, 6, 7}],
+        7: [{7}],
+    }
+    quorums = list(enumerate_quorums(slices_by_node))
+    assert set(quorums) == set(
+        [frozenset({7}),
+         frozenset({4, 5, 6, 7}),
+         frozenset({1, 2, 3, 7}),
+         frozenset({1, 2, 3, 4, 5, 6, 7})]
```

Comment:
This is example 2.8 (from the paper draft). I'd be curious to know how this performs with actual large FBASs.

Reply:
We'll find out! :sunglasses: